Mathlib/MeasureTheory/Integral/Marginal.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2023 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Heather Macbeth
+-/
+import Mathlib.MeasureTheory.Constructions.Pi
+import Mathlib.MeasureTheory.Integral.Lebesgue
+
+/-!
+# Marginals of multivariate functions
+
+In this file, we define a convenient way to compute integrals of multivariate functions, especially
+if you want to write expressions where you integrate only over some of the variables that the
+function depends on. This is common in induction arguments involving integrals of multivariate
+functions.
+This constructions allows working with iterated integrals and applying Tonelli's theorem
+and Fubini's theorem, without using measurable equivalences by changing the representation of your
+space (e.g. `((ι ⊕ ι') → ℝ) ≃ (ι → ℝ) × (ι' → ℝ)`).
+
+## Main Definitions
+* If `f : α → ℝ≥0∞` then `lmarginal μ s f` or `∫⋯∫⁻_s, f ∂μ` is the function that integrates `f`
+  over all variables in `s`. It returns a function that still takes the same variables as `f`,
+  but is constant in the variables in `s`. Mathematically, if `s = {i₁, ..., iₖ}`,
+  then `lmarginal μ s f` is the expression
+  $$
+  \vec{x}\mapsto \int\!\!\cdots\!\!\int f(\vec{x})dx_{i_1}\cdots dx_{i_k}.
```

Comment:
I think this expression is not very clear, as `\vec{x}` is not the same on the left and on the right if I understand correctly.